purchase_sale_inter_company/models/stock_picking.py

```diff
@@ -106,6 +106,23 @@ def button_validate(self):
                         dest_company,
                         record.sale_id,
                     )
+        # if the flag is set, block the validation of the picking in the destination company
+        if self.env.company.block_po_manual_picking_validation:
+            for record in self.sudo():
+                dest_company = record.partner_id.commercial_partner_id.ref_company_ids
+                if (
+                    dest_company
+                    and record.picking_type_code == "incoming"
+                    and record.state == "assigned"
+                    or record.state == "done"
+                    or record.state == "waiting"
```

Comment:
**question**: If I'm understanding this correctly you mean `(dest_company AND picking_type_code AND state == "assigned") OR state == "done" OR state == "waiting"`

And not `(dest_company AND picking_type_code) AND state == ("assigned" OR "done" OR "waiting")`.
Right?


If so I would suggest to add brackets to help visualize the logical expression

```suggestion
                    (
                        dest_company
                        and record.picking_type_code == "incoming"
                        and record.state == "assigned"
                    )
                    or record.state in ["done", "waiting"]
```

Reply:
My intent was the second one, I have rewritten the expression based on your suggestion.